packages/flutter/lib/src/widgets/_html_element_view_web.dart

```diff
@@ -38,9 +39,10 @@ extension HtmlElementViewImpl on HtmlElementView {
   /// The implementation on Flutter Web builds an HTML platform view and handles
   /// its lifecycle.
   Widget buildImpl(BuildContext context) {
+    final FlutterView flutterView = View.of(context);
     return PlatformViewLink(
       viewType: viewType,
-      onCreatePlatformView: _createController,
+      onCreatePlatformView: (PlatformViewCreationParams params) => _createController(params, flutterView.viewId),
```

Comment:
I prefer to keep it as a lambda TBH, but I'll change it to a block syntax (instead of arrow function). The `surfaceFactory` parameter is also being passed as a lambda right below this.

Reply:
I'm good with the lambda :P